hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1218,6 +1219,32 @@ private void finishActiveMasterInitialization(MonitoredTask status) throws IOExc
       LOG.debug("Balancer post startup initialization complete, took " + (
           (EnvironmentEdgeManager.currentTime() - start) / 1000) + " seconds");
     }
+
+    /*
+      A background thread to update all TableDescriptors that do not contain
+      StoreFileTracker implementation configuration. see HBASE-26263.
+    */
+    Thread STFChecking = new Thread(() -> {
+      try {
+        tableDescriptors.getAll().forEach((tableName, htd) -> {
```

Comment:
We may have race here? What if there is a ModifyTableProcedure running at the same time...